### PR TITLE
[Big Bang] Introduce code template

### PR DESCRIPTION
**Pull request by babsey**
This PR adds templates for simulation code.

It also uses getter for all is[A-Z].* and has[A-Z].* functions.

### CODE REVIEW

**Review by jens-br — CHANGES_REQUESTED** [review]
Good additions! I have some suggestions below.

It could also be helpful to refactor the methods `get someParams()` (e.g. to `get hasParams()`) to use a more intuitive name (used in `connection.ts`, `node.ts` and `synapse.ts`).

**Comment by jens-br** [comment]
(renamed to improve the readability of the Git logs and the change list for new versions)

File: src/core/simulation/simulationCode.ts [hunk 1]
```diff
@@ -1,29 +1,165 @@
...
+import Mustache from 'mustache';
+
 import { Simulation } from './simulation';
 
-export class SimulationCode extends Code {
+export class SimulationCode {
+  private _hash: string;
+  private _script: string;
   private _simulation: Simulation; // parent
+  private _state: UnwrapRef<any>;
```
[the remark this answers is not in the document]
Comment (babsey) [reply src/core/simulation/simulationCode.ts:11]: Nice!

File: src/components/model/ModelExplorer.vue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div class="modelExplorer" v-if="modelView.state.model">
     <ActivityGraph
-      :codeHash="modelView.state.project.code.hash"
+      :codeHash="modelView.state.project.simulation.code.hash"
       :graph="modelView.state.project.activityGraph"
       :graphHash="modelView.state.project.activityGraph.codeHash"
       v-if="modelView.state.project"
```
[the remark this answers is not in the document]
Comment (babsey) [reply src/components/model/ModelExplorer.vue:7]: Strange! I cannot reproduce some error showing this notification.

[the remark this answers is not in the document]
Comment (jens-br) [reply src/components/model/ModelExplorer.vue:7]: It is a message in VS Code. We can leave it for now and have a look on it later on.

**Review by jens-br — APPROVED** [review]
LGTM :+1:

**Pull request merged**